### PR TITLE
Refactored Script Engine naming

### PR DESCRIPTION
**Pull request by droyad**
- Removed `IScriptEngine` interface from `CombinedScriptEngine` and replaced it with it's own interface because it should never be confused with the other script engines. It's actually the co-ordinator of the script engines
- Renamed `ScriptEngine` to `ScriptExecutor` to make it clearer what it does and differentiate it from the `CombinedScriptEngine`. The `GetSupportedTypes` method was only used on `CombinedScriptEngine`
- Removed `ScriptEngineRegistry` as it's overkill for the one place it's used

Each step above is a commit and relatively isolated from each other, so easiest to review commit by commit.

### CODE REVIEW

File: source/Calamari.Shared/Integration/Packages/Download/DockerImagePackageDownloader.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace Calamari.Integration.Packages.Download
 {
     public class DockerImagePackageDownloader : IPackageDownloader
     {
-        readonly IScriptEngine scriptEngine;
+        readonly ICombinedScriptEngine scriptEngine;
```
Comment (michaelnoonan) [inline source/Calamari.Shared/Integration/Packages/Download/DockerImagePackageDownloader.cs]: Worth renaming the field as well?

File: source/Calamari/Program.cs [hunk 2]
```diff
@@ -74,7 +74,7 @@ protected virtual ContainerBuilder BuildContainer(CommonOptions options)
             builder.RegisterInstance(fileSystem).As<ICalamariFileSystem>();
             builder.RegisterType<VariablesFactory>().AsSelf();
             builder.Register(c => c.Resolve<VariablesFactory>().Create(options)).As<IVariables>().SingleInstance();
-            builder.RegisterType<CombinedScriptEngine>().AsSelf().As<IScriptEngine>();
+            builder.RegisterType<CombinedScriptEngine>().AsSelf().As<ICombinedScriptEngine>();
```
Comment (michaelnoonan) [inline source/Calamari/Program.cs]: Does it need the AsSelf() registration?

File: source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs [hunk 3]
```diff
@@ -74,5 +91,24 @@ public ScriptSyntax[] GetSupportedTypes()
                      */ 
                     return next;
                 });
+
+        IScriptExecutor GetScriptExecutor(ScriptSyntax scriptSyntax)
```
Comment (michaelnoonan) [inline source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs]: Yes.

File: source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs [hunk 4]
```diff
@@ -74,5 +91,24 @@ public ScriptSyntax[] GetSupportedTypes()
...
+                case ScriptSyntax.CSharp:
+                    return new ScriptCSScriptExecutor();
+                case ScriptSyntax.Bash:
+                    return new BashScriptExecutor();
+                case ScriptSyntax.FSharp:
+                    return new FSharpExecutor();
+                case ScriptSyntax.Python:
+                    return new PythonScriptExecutor();
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(scriptSyntax), scriptSyntax, null);
```
Comment (michaelnoonan) [inline source/Calamari.Shared/Integration/Scripting/CombinedScriptEngine.cs]: Or  `NotSupportedException`.

File: source/Calamari.Shared/Integration/Scripting/FSharp/FSharpExecutor.cs [hunk 5]
```diff
@@ -5,13 +5,8 @@
 
 namespace Calamari.Integration.Scripting.FSharp
 {
-    public class FSharpEngine : ScriptEngine
+    public class FSharpExecutor : ScriptExecutor
     {
-        public override ScriptSyntax[] GetSupportedTypes()
```
Comment (michaelnoonan) [inline source/Calamari.Shared/Integration/Scripting/FSharp/FSharpExecutor.cs]: Yes!

File: source/Calamari.Shared/Integration/Packages/Download/DockerImagePackageDownloader.cs (hunk 1 above)
Comment (droyad) [reply source/Calamari.Shared/Integration/Packages/Download/DockerImagePackageDownloader.cs]: I'm going the other way, with the 2nd refactor nothing else is a `ScriptEngine`, so renamed `CombinedScriptEngine` to that.

File: source/Calamari/Program.cs (hunk 2 above)
Comment (droyad) [reply source/Calamari/Program.cs]: It did, but doesn't with the latest commit

**Pull request merged**